File: cal.py
"""
Python Implementation
"""

import time
import math

def fibo(n):
  if n < 2:
    return n
  else:
    return fibo(n-2) + fibo(n-1)

start = time.time()
result = fibo(40)
finish = time.time()

print('fibo', result, 'in python runs', int((finish - start)*1000))

def slow(n):
  result = 0
  invN = 1 / n
  i = 0
  while (i < n):
    x = math.pow(i * invN, 7)
    result += math.atan(x) * math.tan(x)
    i += 1
  return result

start = time.time()
result = slow(10000000)
finish = time.time()

print('slow', result, 'in python runs', int((finish - start)*1000))

File: cal.js
'use strict'

/**
 * JavaScript implementation
 */
function fibo (n) {
  if (n < 2) {
    return n
  } else {
    return fibo(n - 2) + fibo(n - 1)
  }
}

function slow (n) {
  let result = 0
  const invN = 1 / n
  let i = 0
  while (i < n) {
    const x = Math.pow(i * invN, 7)
    result += Math.atan(x) * Math.tan(x)
    i++
  }
  return result
}

let start = Date.now()
let result = fibo(40)
let finish = Date.now()
console.log('fibo', result, 'in js runs', finish - start)

start = Date.now()
result = slow(10000000)
finish = Date.now()
console.log('slow', result, 'in js runs', finish - start)
